lib/utils.js

```diff
@@ -87,6 +87,8 @@ exports.GreatResponse = function (request, options, filterRules) {
 
     var applyFilter = function (data) {
 
+        if(!data || Object.keys(data).length === 0) return;
```

Comment:
I added also the condition `Object.keys(data).length > 0`, as empty objects also seem to cause this bug.